src/index.js

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable max-classes-per-file */
+/* eslint-disable  import/no-cycle */
+
+import './style.css';
+import interactive from './interactiveList.js';
+import clearCompletedTasks from './addRemove.js';
+
+class SingleToDo {
+  constructor(description) {
+    this.description = description;
+    this.completed = false;
+    this.index = 0;
+  }
+}
+class ToDoList {
+  constructor(toDoTasksArray = [], container) {
+    this.toDoTasksArray = toDoTasksArray;
+    this.container = document.querySelector(container);
+  }
+
+  addToDo(todo) {
+    const newToDo = new SingleToDo(todo);
+    this.toDoTasksArray.push(newToDo);
+    // update index
+    this.toDoTasksArray = this.toDoTasksArray.map((todo, index = 1) => {
+      todo.index = index;
+      return todo;
+    });
+    this.displayToDo();
+    this.setListToLocal(this.toDoTasksArray);
+    window.location.reload();
+  }
+
+  removeToDo(todoId) {
+    const filterToDo = this.toDoTasksArray.filter((todo) => parseInt((todoId), 10) !== todo.index);
+    this.toDoTasksArray = filterToDo;
+    // update index
+    this.toDoTasksArray = this.toDoTasksArray.map((todo, index = 1) => {
+      todo.index = index;
+      return todo;
+    });
+    this.displayToDo();
+    this.setListToLocal(this.toDoTasksArray);
+    window.location.reload();
+  }
+
+  setToDoArray(newToDoArray) {
+    this.toDoTasksArray = newToDoArray;
+    this.setListToLocal(this.toDoTasksArray);
+  }
+
+  getToDoArray() {
+    return this.toDoTasksArray;
+  }
+
+  setListToLocal() {
+    localStorage.setItem('lists', JSON.stringify(this.toDoTasksArray));
+  }
+
+  getListFromLocal() {
+    const getList = localStorage.getItem('lists');
+    if (getList) {
+      this.toDoTasksArray = JSON.parse(getList);
+    }
+    this.displayToDo();
+  }
+
+  displayToDo() {
+    this.toDoTasksArray.sort((a, b) => a.index - b.index);
```

Comment:
Maybe here you don't need this Sort method because when you add or delete you actually reset the index